rdwatch/core/utils/worldview_nitf/raster_tile.py

```diff
@@ -9,12 +9,39 @@
 import boto3
 import rasterio  # type: ignore
 from rio_tiler.io.rasterio import Reader
+from rio_tiler.models import ImageData
+from rio_tiler.utils import pansharpening_brovey
 
 from rdwatch.core.utils.worldview_nitf.satellite_captures import WorldViewNITFCapture
 
 logger = logging.getLogger(__name__)
 
 
+def find_rgb_channels(colorinterp):
+    rgb_indexes = {}
+
+    # Define the channels to look for
+    channels = ['red', 'green', 'blue', 'gray']
+
+    for index, channel in enumerate(colorinterp):
+        if channel in channels:
+            rgb_indexes[channel] = index
+
+    # Return indexes as a tuple, band indexes are 1 BASED
+    red = rgb_indexes.get('red', -1) + 1
+    green = rgb_indexes.get('green', -1) + 1
+    blue = rgb_indexes.get('blue', -1) + 1
+    gray = rgb_indexes.get('gray', -1) + 1
+    if red > 0 and green > 0 and blue > 0:
+        return (red, green, blue)
+    elif gray > 0:
+        return gray
+    else:
+        raise ValueError(
+            f'colorInterp: {colorinterp} for files does not have RGB or Gray channels'
+        )
+
+
 @contextmanager
 def _download_nitf_image(uri: str) -> Generator[str, None, None]:
```

Comment:
Can this function be safely removed?

Reply:
yeah I think we can unless if we're going to support any compressed files in the future.